conans/client/build/compiler_flags.py

```diff
@@ -144,16 +174,17 @@ def build_type_define(build_type=None):
     returns definitions specific to the build type (Debug, Release, etc.)
     like DEBUG, _DEBUG, NDEBUG
     """
-    return 'NDEBUG' if build_type == 'Release' else ""
+    return 'NDEBUG' if build_type in ['Release', 'RelWithDebInfo', 'MinSizeRel'] else ""
```

Comment:
the previous code had a bug. but `RelWithDebInfo` and `MinSizeRel` are not that popular, so probably it was never noticed.